src/python/dxpy/scripts/dx.py

```diff
@@ -3785,6 +3785,158 @@ def publish(args):
     except:
         err_exit()
 
+def archive(args):
+    def send_archive_request(target_project, request_input, request_func):
+        api_errors = [InvalidState, ResourceNotFound, PermissionDenied]
+        try:
+            res = request_func(target_project, request_input)
+        except Exception as e:            
+            eprint("Failed request: {}".format(request_input))
+            if type(e) in api_errors:
+                eprint("     API error: {}. {}".format(e.name, e.msg))
+            else: 
+                eprint("     Unexpected error: {}".format(format_exception(e)))
+            
+            err_exit("Failed request: {}. {}".format(request_input, format_exception(e)), code=3)
+        return res              
+
+    def get_valid_archival_input(args, target_files, target_folder, target_project):
+        request_input = {}
+        if not target_project:
+            err_exit('No target project has been set. Please check the input or check your permission to the given project.', code=3)
+        if target_files and target_folder:
+            err_exit('Expecting either a single folder or a list files for each API request', code=3)
+        elif target_files: 
+            target_files = list(target_files)
+            request_input = {"files": target_files}
+        elif target_folder:
+            request_input = {"folder": target_folder, "recurse":args.recurse}
+        else:
+            err_exit("No input file/folder is found in project {}".format(target_project), 
+                            code=3)
+        return request_input
+
+    def get_archival_paths(args):
+        target_project = None
+        target_folder = None
+        target_files = set()
+        
+        paths = [split_unescaped(':', path, include_empty_strings=True) for path in args.path]
+        possible_projects = set()
+        possible_objects = set()
+        for p in paths:
+            if len(p)>2: 
+                err_exit("Path '{}' is invalid. Please check the inputs or check --help for example inputs.".format(p), code=3)
+            elif len(p) == 2:
+                possible_projects.add(p[0])
```

Comment:
```suggestion
                possible_projects.add(p[0])
            elif len(p) == 1:
              possible_projects.add('')
```

Reply:
Fixed: `:file-xxxx` and `file-xxxx` are now assumed to be in selected project.